### PR TITLE
 Enhance metadata of sharepointLoader.

### PR DESCRIPTION
**Pull request by rahul-trip**
2 feature flags added in this PR:
1. `load_auth`: if set to True, adds `authorised identities` to metadata
2. `load_extended_metadata`, adds `source`, `owner` and `full_path` to metadata
of SharePointLoader

### CODE REVIEW

File: libs/community/langchain_community/document_loaders/base_o365.py [hunk 1]
```diff
@@ -169,8 +171,25 @@ def _load_from_object_ids(
...
+                            "created_by": str(file.created_by),
+                            "modified_by": str(file.modified_by),
+                            "description": file.description,
+                            "id": str(file.object_id),
+                        }
+
             loader = FileSystemBlobLoader(path=temp_dir)
-            yield from loader.yield_blobs()
+            for blob in loader.yield_blobs():
+                if not isinstance(blob.path, PurePath):
```
[the remark this answers is not in the document]
Comment (srics) [reply libs/community/langchain_community/document_loaders/base_o365.py:187]: @rahul-trip Is this portion fixing an existing bug unrelated to this PR? If yes, I would recommend to decouple this out and takes it up in a separate PR

[the remark this answers is not in the document]
Comment (rahul-trip) [reply libs/community/langchain_community/document_loaders/base_o365.py:187]: @Raj725 
Nope, I dont think so. Do you see anything worth checking?

@srics,
This is not a bug fix, instead we need id to get populated, hence added these metadata.

File: libs/community/langchain_community/document_loaders/sharepoint.py [hunk 2]
```diff
@@ -77,10 +78,24 @@ def lazy_load(self) -> Iterator[Document]:
             if not isinstance(target_folder, Folder):
                 raise ValueError(f"There isn't a folder with path {self.folder_path}.")
             for blob in self._load_from_folder(target_folder):
-                yield from blob_parser.lazy_parse(blob)
+                if self.load_auth is True:
+                    for parsed_blob in blob_parser.lazy_parse(blob):
+                        file_id = blob.metadata.get("id")
+                        auth_identities = self.authorized_identities(file_id)
```
[the remark this answers is not in the document]
Comment (rahul-trip) [reply libs/community/langchain_community/document_loaders/sharepoint.py]: Agreed and fixed.

**Review by Raj725 — APPROVED** [review]

**Comment by Raj725** [comment]
Closing this PR since the changes are already merged into `langchain-ai/master` via [PR-22248](https://github.com/langchain-ai/langchain/pull/22248)

**Pull request closed without merging**